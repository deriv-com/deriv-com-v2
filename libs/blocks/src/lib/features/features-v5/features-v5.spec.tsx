import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import FeaturesV4 from '.';
import { Button, Heading } from '@deriv/quill-design';

const content = <Heading.H3>content</Heading.H3>;
const title = 'this is title';
const description = 'this is description';

describe('FeaturesV4', () => {
  beforeEach(() => {
    render(
      <FeaturesV4 title={title} description={description} content={content}>
        <Button>Click me</Button>
      </FeaturesV4>,
    );
  });

  it('should rebder heading correctly', () => {
    const heading = screen.getByRole('heading', {
      name: title,
      level: 2,
    });
    expect(heading).toBeInTheDocument();
  });

  it('should render description correctly', () => {
    const desc = screen.getByText(description);
    expect(desc).toBeInTheDocument();
  });

  it('should render children passed to it', () => {
    const button = screen.getByRole('button', { name: 'Click me' });
    expect(button).toBeInTheDocument();
  });

  it('should render content passed to it', () => {
    const data = screen.getByRole('heading', { name: 'content', level: 3 });
    expect(data).toBeInTheDocument();
  });
});

describe('Content Position', () => {
  it('Should render Feature Block correct content left', () => {
    const { container } = render(
      <FeaturesV4
        title={title}
        description={description}
        content={content}
        variant="content-left"
      >
        <Button>Click me</Button>
      </FeaturesV4>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Should render Feature Block correct content right', () => {
    const { container } = render(
      <FeaturesV4
        title={title}
        description={description}
        content={content}
        variant="content-right"
      >
        <Button>Click me</Button>
      </FeaturesV4>,
    );
    expect(container).toMatchSnapshot();
  });
});
