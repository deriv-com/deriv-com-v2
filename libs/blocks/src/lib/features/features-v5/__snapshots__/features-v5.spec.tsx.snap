// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Content Position > Should render Feature Block correct content left 1`] = `
<div>
  <section
    class="w-full py-general-4xl bg-background-primary-container"
  >
    <div
      class="mx-auto w-full px-800 md:px-1200 min-[1232px]:px-general-none flex max-w-[1024px] items-center gap-gap-3xl lg:gap-gap-lg justify-center lg:flex-row flex-col"
    >
      <div
        class="flex"
      >
        <h3
          class="font-heading text-heading-h3 font-bold leading-heading-h3 space-y-paragraphSpacing-heading-h3 text-typography-prominent"
        >
          content
        </h3>
      </div>
      <div
        class="flex flex-1 basis-full flex-col w-full lg:pl-general-2xl"
      >
        <h2
          class="font-heading text-heading-h2 font-bold leading-heading-h2 space-y-paragraphSpacing-heading-h2 text-typography-prominent"
        >
          this is title
        </h2>
        <p
          class="font-regular text-typography-default text-body-md leading-body-md space-y-paragraphSpacing-body-md pt-general-md"
        >
          this is description
        </p>
        <div
          class="pt-general-2xl"
        >
          <button
            class="transition-colors duration-150 ease-in-out font-bold disabled:opacity-600 disabled:cursor-not-allowed flex items-center justify-center cursor-pointer bg-solid-coral-700 text-solid-slate-50 enabled:hover:bg-solid-coral-800 enabled:active:bg-solid-coral-900 rounded-600 text-body-sm leading-body-sm h-[32px] min-w-[64px] px-600 py-300 gap-300"
          >
            <span
              class="flex-1"
            >
              Click me
            </span>
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`Content Position > Should render Feature Block correct content right 1`] = `
<div>
  <section
    class="w-full py-general-4xl bg-background-primary-container"
  >
    <div
      class="mx-auto w-full px-800 md:px-1200 min-[1232px]:px-general-none flex max-w-[1024px] items-center gap-gap-3xl lg:gap-gap-lg justify-center lg:flex-row-reverse flex-col"
    >
      <div
        class="flex"
      >
        <h3
          class="font-heading text-heading-h3 font-bold leading-heading-h3 space-y-paragraphSpacing-heading-h3 text-typography-prominent"
        >
          content
        </h3>
      </div>
      <div
        class="flex flex-1 basis-full flex-col w-full lg:pr-general-2xl"
      >
        <h2
          class="font-heading text-heading-h2 font-bold leading-heading-h2 space-y-paragraphSpacing-heading-h2 text-typography-prominent"
        >
          this is title
        </h2>
        <p
          class="font-regular text-typography-default text-body-md leading-body-md space-y-paragraphSpacing-body-md pt-general-md"
        >
          this is description
        </p>
        <div
          class="pt-general-2xl"
        >
          <button
            class="transition-colors duration-150 ease-in-out font-bold disabled:opacity-600 disabled:cursor-not-allowed flex items-center justify-center cursor-pointer bg-solid-coral-700 text-solid-slate-50 enabled:hover:bg-solid-coral-800 enabled:active:bg-solid-coral-900 rounded-600 text-body-sm leading-body-sm h-[32px] min-w-[64px] px-600 py-300 gap-300"
          >
            <span
              class="flex-1"
            >
              Click me
            </span>
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;
